ko/pl.rst

```diff
@@ -0,0 +1,1383 @@
+            RETURN -1;
+        WHEN OTHERS THEN
+            put_line('unknown exception');
+            RETURN -1;
+    END;
+
+.. _static_sql:
+
+Static SQL
+==================
+
+SQL 구문 중에 다음에 해당하는 것들을 PL/CUBSQL 실행문으로 직접 사용할 수 있으며,
+그러한 경우를 Static SQL 문이라고 부른다.
+
+* SELECT
+* INSERT, UPDATE, DELETE, MERGE
+* COMMIT, ROLLBACK
+
+위 목록에 포함되지 않는 다른 SQL 문들은 직접 사용할 수는 없으나,
+아래에서 설명하는 Dynamic SQL 문을 써서 실행할 수 있다.
+
+SELECT 문은 실행문으로 사용될 뿐만 아니라 커서를 선언할 때나 커서 변수를 OPEN 할 때도 사용된다.
+
+Static SQL 문의 WHERE 절이나 VALUES 절 안에서처럼 값을 필요로 하는 자리에
+프로그램에서 선언한 변수나 프로시저/함수 파라메터를 쓸 수 있다.
+그리고, SELECT 문의 INTO 절에 프로그램의 변수나 OUT 파라메터를 써서 조회 결과를 담을 수 있다.
+
+SQL 구문의 문법과 의미는 CUBRID 매뉴얼 중
+`CUBRID SQL <https://www.cubrid.org/manual/ko/11.2/sql/index.html>`_\을 참고하도록 한다.
+
+.. code-block:: sql
+
+    CREATE FUNCTION get_medal_count(name VARCHAR, medal CHAR) RETURN INTEGER AS
+        n INTEGER;
+    BEGIN
+        -- 일반 실행문으로서의 SELECT 문
+        SELECT COUNT(medal) INTO n FROM athlete a, record r
+            WHERE a.name = name AND a.code = r.athlete_code AND r.medal = medal;
+
+        RETURN n;
+    END;
+
+.. code-block:: sql
+
+    CREATE PROCEDURE athlete_history(name VARCHAR) AS
+    BEGIN
+        -- For 루프 안에서의 SELECT 문
+        FOR r IN (SELECT host_year, score FROM history WHERE athlete = name) LOOP
+            put_line('host_year: ' || r.host_year || ' score: ' || r.score);
+        END LOOP;
+    END;
+
+.. code-block:: sql
+
+    CREATE PROCEDURE athlete_history(name VARCHAR) AS
+        -- 커서 정의에서의 SELECT 문
+        CURSOR my_cursor IS SELECT host_year, score FROM history WHERE athlete = name;
+    BEGIN
+        FOR r IN my_cursor LOOP
+            put_line('host_year: ' || r.host_year || ' score: ' || r.score);
+        END LOOP;
+    END;
+
+.. _dyn_sql:
+
+Dynamic SQL
+==================
+
+Dynamic SQL은 실행 시간에 SQL 구문에 해당하는 문자열을 조합하여
+:ref:`EXECUTE IMMEDIATE <exec_imme>` 문으로 실행하는 방식이다.
+Dynamic SQL은 다음 두 가지 경우에 필요하다.
+
+* 실행하려는 SQL 구문을 프로그램 작성 시에 결정하는 것이 어렵거나 불가능한 경우
+* DDL 문처럼 Static SQL이 지원하지 않는 구문을 실행해야 할 경우
+
+아래 예제에서 새 테이블 이름은 프로시저 인자를 포함하므로 프로그램 작성 시에 결정할 수 없고
+프로그램 실행 시간에야 결정되는 값이다.
+그리고, DROP TABLE 문과 CREATE TABLE 문은 Static SQL 기능에서 지원하지 않는 DDL 문이다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE collect_athlete_history(name VARCHAR) AS
+        new_table VARCHAR := name || '_history';
+    BEGIN
+        EXECUTE IMMEDIATE 'drop table if exists ' || new_table;
+        EXECUTE IMMEDIATE 'create table ' || new_table || ' like history';
+        EXECUTE IMMEDIATE 'insert into ' || new_table || ' select * from history where athlete = ?'
+            USING name;
+    END;
+
+작성 규칙
+==================
+
+식별자, 예약어, 주석, 리터럴을 작성할 때 Static/Dynamic SQL 안에서는
+`CUBRID SQL의 작성 규칙 <https://www.cubrid.org/manual/ko/11.2/sql/syntax.html>`_\을 따른다.
+
+Static/Dynamic SQL 밖의 PL/CUBSQL 문 작성 규칙도 대체로 같은 규칙을 따르지만 다음 몇 가지 예외가 있다.
+
+* CUBRID SQL과 달리 식별자에 '#'을 쓸 수 없다. 즉, 식별자는 영문 대소문자, 숫자, '_'(underscore)로만 이루어져야 한다.
+* 큰따옴표, 대괄호, 백틱 부호로 둘러싸더라도 식별자에 특수 문자를 쓸 수 없다.
```

Comment:
To use PL/CUBSQL, users cannot create tables with special characters in their names.

Isn't this the same as disabling special characters from "CUBRID SQL > 작성 규칙 > 식별자"?
At least, it means that users cannot use tables with special characters in their names in static sql.

Reply:
In Static/Dynamic SQL, the same lexical rules as CUBRID SQL are applied and users can use special characters with appropriate delimiters (" ", [ ], ` `). Therefore, the above [table-1] is valid.
Outside Static/Dynamic SQL, users cannot use special characters even with the delimiters. This is because they are converted to Java identifiers.